ovs/dal/migration/albamigrator.py

```diff
@@ -126,17 +141,135 @@ def migrate(previous_version):
                 for alba_disk in alba_node.disks:
                     if alba_disk.aliases is not None:
                         continue
-                    if 'name' in alba_disk._data:
-                        alba_disk.aliases = ['/dev/disk/by-id/{0}'.format(alba_disk._data['name'])]
+                    # noinspection PyProtectedMember
+                    alba_disk_data = alba_disk._data
+                    if 'name' in alba_disk_data:
+                        alba_disk.aliases = ['/dev/disk/by-id/{0}'.format(alba_disk_data['name'])]
                         alba_disk.save()
 
             # Rename of arakoon services
-            from ovs.dal.hybrids.servicetype import ServiceType
-            from ovs.dal.lists.servicelist import ServiceList
             for service in ServiceList.get_services():
                 if service.type.name == ServiceType.SERVICE_TYPES.ALBA_MGR or service.type.name == ServiceType.SERVICE_TYPES.NS_MGR:
                     if not service.name.startswith('arakoon-'):
                         service.name = 'arakoon-{0}'.format(service.name)
                         service.save()
 
+            #############################################
+            # Introduction of ABMCluster and NSMCluster #
+            #############################################
+            # Retrieve ABM and NSM clusters
+            abm_cluster_info = []
+            nsm_cluster_info = []
+            for cluster_name in Configuration.list('/ovs/arakoon'):
+                try:
+                    metadata = ArakoonInstaller.get_arakoon_metadata_by_cluster_name(cluster_name=cluster_name)
+                    if metadata['cluster_type'] == ServiceType.ARAKOON_CLUSTER_TYPES.ABM:
```

Comment:
At this point, already check the model. If that cluster is already migrated, there's no need to execute all below logic for that cluster.